private-templates-service/client/src/components/NavBar/NavBar.tsx

```diff
@@ -1,90 +1,42 @@
-import React, { ReactElement } from "react";
+import React from "react";
 import {
   Collapse,
   Container,
   Navbar,
   NavbarToggler,
   NavbarBrand,
-  Nav,
-  NavItem,
-  NavLink,
-  UncontrolledDropdown,
-  DropdownToggle,
-  DropdownMenu,
-  DropdownItem
+  Nav
 } from "reactstrap";
 
+// Redux
+import { connect } from "react-redux";
+import { RootState } from "../../store/rootReducer";
+
 // Components
-import { UserType } from '../../App';
+import { UserType } from "../../store/auth/types";
+import AuthNavItem from "./AuthNavItem";
 
 // Styles
 import "@fortawesome/fontawesome-free/css/all.css";
-import { AvatarIcon, DefaultAvatarIcon } from './styled';
-
-function UserAvatar(props: { user?: UserType }): ReactElement {
-  // If a user avatar is available, return an img tag with the pic
-  if (props.user && props.user.avatar) {
-    return (
-      <AvatarIcon
-        src={props.user.avatar}
-        alt="user"
-        className="rounded-circle align-self-center mr-2"
-      ></AvatarIcon>
-    );
-  }
 
-  // No avatar available, return a default icon
-  return (
-    <DefaultAvatarIcon
-      className="far fa-user-circle fa-lg rounded-circle align-self-center mr-2"
-    ></DefaultAvatarIcon>
-  );
-}
-
-function AuthNavItem(props: NavBarProps): ReactElement {
-  // If authenticated, return a dropdown with the user's info and a
-  // sign out button
-  if (props.isAuthenticated && props.user) {
-    return (
-      <UncontrolledDropdown>
-        <DropdownToggle nav caret>
-          <UserAvatar user={props.user} />
-        </DropdownToggle>
-        <DropdownMenu right>
-          <h5 className="dropdown-item-text mb-0">
-            {props.user.displayName}
-          </h5>
-          <p className="dropdown-item-text text-muted mb-0">
-            {props.user.email}
-          </p>
-          <DropdownItem divider />
-          <DropdownItem onClick={props.authButtonMethod}>
-            Sign Out
-					</DropdownItem>
-        </DropdownMenu>
-      </UncontrolledDropdown>
-    );
-  }
-
-  // Not authenticated, return a sign in link
-  return (
-    <NavItem>
-      <NavLink onClick={props.authButtonMethod}>Sign In</NavLink>
-    </NavItem>
-  );
-}
+const mapStateToProps = (state: RootState) => {
+  return {
+    isAuthenticated: state.auth.isAuthenticated,
+    user: state.auth.user
+  };
+};
 
 export interface NavBarProps {
-  isAuthenticated: boolean,
-  authButtonMethod: () => void,
-  user?: UserType,
+  authButtonMethod: () => void;
```

Comment:
I looked into this, it seems to be used interchangeably, typescript docs use semicolons but sites like tutorialpoint use commas. I will leave it as is since it seems like either will work the same.